### ISSUE TITLE
[Feature Request] Explore Higher Level APIs in MSAL 

### ISSUE THREAD

**Issue by gladjohn** [AzureAD/microsoft-authentication-library-for-dotnet#4153]
Explore commonly used MSAL Auth scenarios and see how these different API calls can be abstracted into an easier way to call into an Higher Level API

**Comment by bgavrilMS** [comment]
This is somewhat vague @gladjohn. Id.Web does most of the higher level API stuff. 

In my opinion, anything that other SDKs and power-users need to use should be in MSAL. The rest in ID.Web.

**Comment by jmprieur** [comment]
Would that we for PCA?
cc: @localden

**Comment by localden** [comment]
Yes, I think for PCA we should invest some time in assessing whether there is a customer need for a higher-level API. Especially when we start thinking about scenarios like CAE (thank you @jmprieur for clarity there), this goes a bit out of the MSAL territory. Will assign this item to myself for now.
